apache2/msc_reqbody.c

```diff
@@ -428,55 +428,62 @@ apr_status_t modsecurity_request_body_store(modsec_rec *msr,
 }
 
 apr_status_t modsecurity_request_body_to_stream(modsec_rec *msr, const char *buffer, int buflen, char **error_msg) {
-    char *stream_input_body = NULL;
-    char *data = NULL;
-    int first_pkt = 0;
+    apr_size_t allocate;
+    char* allocated;
 
-    if(msr->stream_input_data == NULL)  {
-        msr->stream_input_data = (char *)calloc(sizeof(char), msr->stream_input_length + 1);
-        first_pkt = 1;
-    }
-    else    {
-
-        data = (char *)malloc(msr->stream_input_length + 1 - buflen);
+    if (msr->stream_input_data == NULL)  {
+        // Is the request body length is known beforehand? (requests that are not Transfer-Encoding: chunked)
+        if (msr->request_content_length > 0) {
+            allocate = msr->request_content_length;
```

Comment:
that's true, they used to null terminate. Maybe they were considering treating the buffer as a c string at some point, but the buffer can contain binary data, so that wouldn't make sense. I haven't found anywhere where the buffer is used as a string without the length. I shall get rid of the +1 use over in re_operators.c for consistency.